### PR TITLE
Add initial cube download script

### PR DESCRIPTION
**Pull request by Frost36**
This greasemonkey script will download the primary data for a cube from the Edit Cube page on Cube Tutor and save it into a TOML file.

Example Output:

[Abhorrent Overlord]
mana_cost = ["5", "B", "B"]
color_identity = "B"
types = ["creature"]
tags = ["Reanimator - Payoff","Event - Sudden","Big","Keyword","ETB - Enabler"]

### CODE REVIEW

File: data/cube_downloader.js [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+/**
+ * Converts a DOM node containing a mana symbol into a string representing the symbol.
+ *
+ * @param {Element} symbolNode The DOM element containing a single mana symbol.
+ * @return {string} The string representing the symbol.
+ */
+function convertManaSymbol(symbolNode) {
+    var symbol = symbolNode.className;
+    symbol = symbol.replace("symbol", "");
```
Comment (pseudomelon) [inline data/cube_downloader.js]: This code will break as soon as Wizards adds the SymbolSymbol manatype ;)

File: data/cube_downloader.js [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+/**
+ * The main function of this script, which gets run on execution.
+ * Appends a button to the menu that will parse out and download the cube list into a file.
+ */
+(function() {
+    var buttonNode = document.createElement('div');
+    buttonNode.innerHTML = '<button id="' + BUTTON_ID + '" type="button">Download Card Tags</button>';
+    buttonNode.setAttribute ('id', BUTTON_CONTAINER_ID);
+
+    document.getElementById('accordion_0').appendChild(buttonNode);
```
[the remark this answers is not in the document]
Comment (Frost36) [reply data/cube_downloader.js]: No, I think that's a reasonable thing to do.

File: data/cube_downloader.js [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+    return selector.options[selector.selectedIndex];
+}
+
+/**
+ * Converts an array to a parseable printed string.
+ *
+ * @param {Array} array The array to stringify.
+ * @return {string} The stringified array
+ */
+function arrayToString(array) {
```
[the remark this answers is not in the document]
Comment (Frost36) [reply data/cube_downloader.js]: At first I didn't think so, but then I realized that this is still effectively just JSON - looks like `JSON.stringify` does what I want.

File: data/cube_downloader.js [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+ * @param {string} content The generated output file content downloaded so far.
+ * @param {number} page The page to download when running this function.
+ * @param {number} numPages The total number of pages to download, so we know when to stop.
+ * @param {string} cubeId The id of the current cube, for generating the URL to POST to.
+ */
+function downloadCardPage(content, page, numPages, cubeId) {
+    // If we're on the last page, save the file with all the content that we have so far.
+    if(page > numPages) {
+        saveFile(content);
+        // Otherwise, load the next page and wait a second before downloading it.
```
Comment (uponthesun) [inline data/cube_downloader.js]: "Wait a second" - stale comment

**Pull request merged**